### PR TITLE
feat(PeriphDrivers): Add max32657 basic files

### PR DESCRIPTION
**Pull request by ozersa**
### Description
MAX32657 is a new MCU that derived from MAX32655 SoC.
This PR add CMakefiles and basic MAX32657 includes files.
Peripheral drivers files are exactly same as MAX32655, just MAX32655 replaced with MAX32657, 
These files added here to demonstrate delta clearly on next commits.
On next commit the file content will be updated incase of need.

CMSIS files is going to be added by an individual PR.

File List:
image

### Checklist Before Requesting Review

- [ ] PR Title follows correct guidelines.
- [ ] Description of changes and all other relevant information.
- [ ] (Optional) Link any related GitHub issues [using a keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)
- [ ] (Optional) Provide info on any relevant functional testing/validation.  For API changes or significant features, this is not optional.

### CODE REVIEW

File: Libraries/PeriphDrivers/Include/MAX32657/aes.h [hunk 1]
```diff
@@ -0,0 +1,271 @@
+/**
+ * @file
```
Comment (ttmut) [inline Libraries/PeriphDrivers/Include/MAX32657/aes.h:2]: Filename missing.

File: Libraries/PeriphDrivers/Include/MAX32657/i2c.h
Comment (ttmut) [inline Libraries/PeriphDrivers/Include/MAX32657/i2c.h:1]: MAX32657 does not have an I2C block. There is instead an I3C block which can send legacy I2C messages to legacy I2C target devices.

File: Libraries/PeriphDrivers/Include/MAX32657/lpcmp.h
Comment (ttmut) [inline Libraries/PeriphDrivers/Include/MAX32657/lpcmp.h:1]: LPCMP is not available on MAX32657.

File: Libraries/PeriphDrivers/Include/MAX32657/mxc_assert.h [hunk 2]
```diff
@@ -0,0 +1,92 @@
+/**
+ * @file
```
Comment (ttmut) [inline Libraries/PeriphDrivers/Include/MAX32657/mxc_assert.h:2]: Filename missing.

File: Libraries/PeriphDrivers/Include/MAX32657/mxc_delay.h [hunk 3]
```diff
@@ -0,0 +1,125 @@
+/**
+ * @file
```
Comment (ttmut) [inline Libraries/PeriphDrivers/Include/MAX32657/mxc_delay.h:2]: Filename missing.

File: Libraries/PeriphDrivers/Include/MAX32657/mxc_errors.h [hunk 4]
```diff
@@ -0,0 +1,78 @@
+/**
+ * @file
```
Comment (ttmut) [inline Libraries/PeriphDrivers/Include/MAX32657/mxc_errors.h:2]: Filename missing.

File: Libraries/PeriphDrivers/Include/MAX32657/mxc_lock.h [hunk 5]
```diff
@@ -0,0 +1,76 @@
+/**
+ * @file
```
Comment (ttmut) [inline Libraries/PeriphDrivers/Include/MAX32657/mxc_lock.h:2]: Filename missing.

File: Libraries/PeriphDrivers/Include/MAX32657/rtc.h [hunk 6]
```diff
@@ -0,0 +1,226 @@
+/**
+ * @file
```
Comment (ttmut) [inline Libraries/PeriphDrivers/Include/MAX32657/rtc.h:2]: Filename missing.

File: Libraries/PeriphDrivers/Include/MAX32657/sema.h
Comment (ttmut) [inline Libraries/PeriphDrivers/Include/MAX32657/sema.h:1]: Semaphore block does not exist on MAX32657.

File: Libraries/PeriphDrivers/Include/MAX32657/simo.h
Comment (ttmut) [inline Libraries/PeriphDrivers/Include/MAX32657/simo.h:1]: SIMO does not exist.

File: Libraries/PeriphDrivers/Include/MAX32657/spi.h [hunk 7]
```diff
@@ -0,0 +1,703 @@
+/**
+ * @file
```
Comment (ttmut) [inline Libraries/PeriphDrivers/Include/MAX32657/spi.h:2]: Filename missing.

File: Libraries/PeriphDrivers/Source/AES/aes_me30.c [hunk 8]
```diff
@@ -0,0 +1,160 @@
+/**
+ * @file
```
Comment (ttmut) [inline Libraries/PeriphDrivers/Source/AES/aes_me30.c:2]: Filename missing.

File: Libraries/PeriphDrivers/Source/SEMA/sema_me30.c
Comment (ttmut) [inline Libraries/PeriphDrivers/Source/SEMA/sema_me30.c:1]: SEMA does not exist.

File: Libraries/PeriphDrivers/Source/SIMO/simo_me30.c
Comment (ttmut) [inline Libraries/PeriphDrivers/Source/SIMO/simo_me30.c:1]: SIMO does not exist.

File: Libraries/PeriphDrivers/Source/SYS/pins_me30.c [hunk 9]
```diff
@@ -0,0 +1,168 @@
+/**
+ * @file    mxc_pins.c
```
Comment (ttmut) [inline Libraries/PeriphDrivers/Source/SYS/pins_me30.c:2]: Wrong filename.

File: Libraries/zephyr/MAX/CMakeLists.txt [hunk 10]
```diff
@@ -60,3 +60,4 @@ add_subdirectory_ifdef(CONFIG_SOC_MAX32680 Source/MAX32680)
 add_subdirectory_ifdef(CONFIG_SOC_MAX32690 Source/MAX32690)
 add_subdirectory_ifdef(CONFIG_SOC_MAX78000 Source/MAX78000)
 add_subdirectory_ifdef(CONFIG_SOC_MAX78002 Source/MAX78002)
+add_subdirectory_ifdef(CONFIG_SOC_MAX32657 Source/MAX32657)
```
Comment (ttmut) [inline Libraries/zephyr/MAX/CMakeLists.txt:63]: Should this line be moved upper to make the order of MCUs alphanumeric?

**Review by ttmut — CHANGES_REQUESTED** [review]

File: Libraries/zephyr/MAX/CMakeLists.txt (hunk 10 above)
Comment (ozersa) [reply Libraries/zephyr/MAX/CMakeLists.txt]: Thanks, all them addressed, 
I2C related case shall be handled while working on drivers.

**Review by Jake-Carter — CHANGES_REQUESTED** [review]
@ozersa  Can you target the [feat/ME30](https://github.com/analogdevicesinc/msdk/tree/feat/ME30) branch for this PR?  I added some other placeholder files as well to start the ME30 bare-metal support.  Since this adds the initial PeriphDrivers we should consolidate to that branch, and then we can do another PR to merge it to main

@sihyung-maxim is also adding in the register files on that branch

**Comment by ozersa** [comment]
@Jake-Carter @sihyung-maxim  there are multiple confliction with feat/me30 branch. Can we divide it as below sub-tasks?
and process them one by one.

1- Add basic Wrappers/Drivers files
2- Add register definitions
3- Update peripheral drivers files
4- Add bare-metal support
5- Add BSP package

This PR focus on #1.

**Review by Jake-Carter — APPROVED** [review]

**Pull request merged**